src/mpid/ch4/src/ch4_progress.h

```diff
@@ -184,11 +184,16 @@ MPL_STATIC_INLINE_PREFIX void MPIDI_progress_state_init(MPID_Progress_state * st
     }
 
     state->progress_made = 0;
-    /* global progress by default */
-    for (int i = 0; i < MPIDI_global.n_vcis; i++) {
-        state->vci[i] = i;
+    if (!MPIDI_global.is_initialized) {
```

Comment:
Is there any concern about these new branches in progress? Probably once MPI is warmed up there's not much overhead. Or a `likely()` might be a good idea.

Reply:
My hypothesis is because this condition, once stabilized, never changes, the CPU branch prediction -- if it worth any salt -- should have no issue predict correctly. Thus, I would rather let the CPU do its job and not to clutter the code.

Reply:
Good enough for me.